Automattic/jetpack
# Debugger: Introduce testing suite

Introduces an "unit testing" suite to Jetpack to check for local conditions known to be problematic. Currently adds a check for the master user existing and still an admin, plus refactoring most the existing tests from the debugger into the new format.

Fixes #10249
#### Changes proposed in this Pull Request:
* Adds a "debugger" library to be a canonical source of truth for local testing of the JP connection.
* Refactors the WP CLI `wp jetpack status` check, the REST API `test-connection` endpoint and the in-plugin debugger (admin.php?page=jetpack-debugger) to use the new testing suite.
* Adds a test-wpcom endpoint that encrypts the result of the debug in order to be consumed by WordPress.com.

#### Testing instructions:
* Check out the in-plugin debugger, make sure tests pass on a regularly connected site. Try on a disconnected site, make sure a test fails.
* Using a different admin user than the one that connected Jetpack, downgrade or delete the connecting admin. Check out the debugger again.
* Check both situations above via `wp jetpack status` and see passing/failing results.
* Only using JN sites: Check the jetpack.com debugger to confirm that a working site passes or a site failing a local test shows the local failure for SA a12s only.

#### Proposed changelog entry for your changes:
* Enhancement: Add additional tests to check when Jetpack isn't working as expected and ensures all current debugging platforms use the same testing list.
